ksqldb-common/src/main/java/io/confluent/ksql/properties/PropertiesUtil.java

```diff
@@ -159,4 +163,47 @@ private static void throwOnBlackListedProperties(final Map<String, ?> properties
     props.stringPropertyNames().forEach(key -> builder.put(key, props.getProperty(key)));
     return builder.build();
   }
+
+  public static Map<String, Object> coerceTypes(
+      final Map<String, Object> streamsProperties,
+      final boolean ignoreUnresolved
+  ) {
+    if (streamsProperties == null) {
+      return Collections.emptyMap();
+    }
+
+    final Map<String, Object> validated = new HashMap<>(streamsProperties.size());
+    for (final Map.Entry<String, Object> e : streamsProperties.entrySet()) {
+      try {
+        validated.put(e.getKey(), coerceType(e.getKey(), e.getValue()));
+      } catch (final PropertyNotFoundException p) {
+        if (ignoreUnresolved) {
+          validated.put(e.getKey(), e.getValue());
+        } else {
+          throw p;
+        }
+      }
+    }
+    return validated;
+  }
+
+  private static Object coerceType(final String key, final Object value) {
+    try {
+      final String stringValue = value == null
+          ? null
+          : value instanceof List
+              ? listToString((List<?>) value)
+              : String.valueOf(value);
+
+      return PROPERTY_PARSER.parse(key, stringValue);
+    } catch (final Exception e) {
+      throw new KsqlException("Failed to set '" + key + "' to '" + value + "'", e);
```

Comment:
This exception message is a bit unclear to me, should that be "failed to coerce type of value {} for key {}"?

Reply:
Makes sense - it was just copypasta